pygeo/parameterization/DVGeo.py

```diff
@@ -592,7 +592,11 @@ def addPointSet(self, points, ptName, origConfig=True, **kwargs):
             Flag determine if the coordinates are projected into the
             undeformed or deformed configuration. This should almost
             always be True except in circumstances when the user knows
-            exactly what they are doing."""
+            exactly what they are doing.
+        comm : None
+            Comm that is associated with the added point set. Does not
+            work now, just added to be consistent with the API of
+            other DVGeo types."""
 
```

Comment:
So does the point set have a comm? I am not sure about how pyGeo handles parallelization.
Any plans to fix/improve this at some point?

Reply:
I will mostly defer this question to @anilyil.

To give some context though, the motivation behind adding this is because we want to add `comm` to the `addPointSet` call in ADflow so that DVGeometryMulti can use it. I think we can remove this though because `addPointSet` already takes in arbitrary kwargs and just ignores the ones it doesn't use.

Reply:
The FFD-based DVGeos do not really need a communicator; the only parallel operation is the allreduce after the sensitibity computation, and that method takes in an optional comm. I added the comm to the addpointset call because some of the operations on the pointsets need to be comm-aware. 

There are 2 types of pointsets added to dvgeo: serial (duplicated) or distributed. The "serial" pointsets are identical across all procs and this serial computation is basically duplicated on all procs. When this is done, we dont need a comm, because all procs own the entire copy. These points are added by DVCon, so they would not get the intersection treatment in dvgeo anyways.

The parallel points are what sth like adflow adds. These are distributed among adflow's communicator and each processor gets part (or none) of the points. The FFD-based computation does not need to know the global collection of points, so we don't need a comm except until that final allreduce in the sensitivity computation. 

However, the intersection method does need to know the global collection of nodes. This is because the intersection code will need to be aware of all nodes for a particular step in the algorithm (the curve based projection and warping, sections 2.7 and 2.8 in the paper). Because of this, we are adding this optional argument; the "serial" pointsets dont use it, but the distributed pointsets do use it. On the pyGeo side, only the DVGeoMulti uses it.

Regarding absorbing it into the kwargs; if thats already there then I am fine with it since other dvgeos dont use it and we can leave it explicitly defined in the dvgeomulti code

Reply:
Thanks for the explanation! I see where this goes, but I am still not sure if we want to add a "dead" attribute to this method. I am also wondering if having functions with the same name and "slightly" different behavior in different files (e.g., it looks like we now have three `addPointSet` in pygeo) can cause problems. I assume that once you define the DVGeo object then there are no ambiguities, I am just not sure if this is good practice

Reply:
I also noticed that for consistency you should add `compNames` here. Do we want to add a warning in case these two arguments are passed in here by mistake. something like "method called with XXX attribute, but XXX is not implemented on DVGeo so it will be ignored"

Reply:
I removed `comm` from this function and similarly will not add `compNames`.

For the warning, I think it is rare for the user to call `addPointSet` directly. This is mainly used by DVCon and our solvers. DVCon now passes `compNames` for every point set based constraint, so having a warning every time would be annoying.

As for whether this is good practice, I can't really say. The reason we do it is mainly to reduce code duplication in DVCon. It would be cumbersome to add code to each constraint to check what type of DVGeo was passed in and then use the appropriate `addPointSet` call.